sdk/core/azure-core/src/http/curl/curl_session_private.hpp

```diff
@@ -241,14 +241,14 @@ namespace Azure { namespace Core { namespace Http {
      * are expecting to.
      *
      */
-    int64_t m_contentLength;
+    int64_t m_contentLength = 0;
 
     /**
      * @brief For chunked responses, this field knows the size of the current chuck size server
      * will de sending
      *
      */
-    int64_t m_chunkSize;
+    int64_t m_chunkSize = 0;
 
     int64_t m_sessionTotalRead = 0;
```

Comment:
What about other fields currently being assigned in the ctor like `m_isChunkedResponseType`?

Let's remove the redundant assignments in the ctor body as well like `m_sessionTotalRead` which is initialized in-class:
https://github.com/Azure/azure-sdk-for-cpp/blob/3abecc3fec8630445942441b7ab495b93044027d/sdk/core/azure-core/src/http/curl/curl_session_private.hpp#L345-L352

Reply:
If you do so - don't assign in constructor, use the initializer list:
```cpp
CurlSession(Request& request, std::unique_ptr<CurlNetworkConnection> connection, bool keepAlive) 
    : m_connection(std::move(connection)),
      m_request(request),
      m_keepAlive(keepAlive),
      m_bodyStartInBuffer(-1),
      m_innerBufferSize(Details::c_DefaultLibcurlReaderSize),
      m_isChunkedResponseType(false),
      m_sessionTotalRead(0),
{ } 
```